test/procedure.test.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+var Database = require('../lib/database');
+var drivers = require('./drivers');
+var should = require('should');
+
+var getDriver = function(name, opts) {
+  return new Database(name, opts);
+};
+
+var driver = drivers[1];
+var db = getDriver(driver.name, driver.opts);
+
+before(function(done) {
+  db.execute(driver.dropSP1)
+    .then(function(){
+      db.execute(driver.createSP1);
+    })
+    .then(function(){
+      db.execute(driver.dropSP2);
+    })
+    .then(function(){
+      db.execute(driver.createSP2);
+    })
+    .then(function(){
+      db.execute(driver.dropSP3);
+    })
+    .then(function(){
+      db.execute(driver.createSP3);
+      done();
+    });
+});
+
+after(function(done) {
+  db.execute(driver.dropSP1)
+    .then(function(){
+      db.execute(driver.dropSP2);
+    })
+    .then(function(){
+      db.execute(driver.dropSP3);
+        done();
+    });
+});
+
+describe('Happy Stored Procedure', function() {
+
+  it('should return record set', function(done) {
+      db.procedure.execute('SP_TEST1', function(err, recordSet){
```

Comment:
Because you've changed to a callback earlier, you've now to a confused API, the signature for execute is callback based, when everything else is promise based